tests/e2e/websocket/common.hpp

```diff
@@ -0,0 +1,302 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <algorithm>
+#include <chrono>
+#include <cstdint>
+#include <iostream>
+#include <memory>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <thread>
+#include <utility>
+
+#include <gflags/gflags.h>
+#include <spdlog/spdlog.h>
+#include <unistd.h>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/strand.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/core/buffers_to_string.hpp>
+#include <boost/beast/ssl/ssl_stream.hpp>
+#include <boost/beast/websocket.hpp>
+#include <json/json.hpp>
+#include <mgclient.hpp>
+
+#include "utils/logging.hpp"
+
+namespace beast = boost::beast;
+namespace http = beast::http;
+namespace websocket = beast::websocket;
+namespace net = boost::asio;
+using tcp = boost::asio::ip::tcp;
+namespace ssl = boost::asio::ssl;
+
+constexpr std::array kSupportedLogLevels{"debug", "trace", "info", "warning", "error", "critical"};
+
+struct Credentials {
+  std::string_view username;
+  std::string_view passsword;
```

Comment:
`string_view` works for this example because you use string literals for all your use cases but remember that it can be dangerous as it doesn't increase the lifetime of the `std::string` object.
E.g. if you had something like
```
int i = 1;
auto user = fmt::format("user{}", i);
Credenntials  creds{user, "password");
```
The lifetime of `username` inside the `creds` depends on the `user` variable.